dfusion_rust_core/src/models/solution.rs

```diff
@@ -3,24 +3,30 @@ use super::*;
 use log::info;
 
 use byteorder::{BigEndian, ByteOrder};
+use std::collections::HashMap;
 use std::iter::once;
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct Solution {
-    pub prices: Vec<u128>,
+    // token_id => price
+    pub prices: HashMap<u16, u128>,
     pub executed_buy_amounts: Vec<u128>,
     pub executed_sell_amounts: Vec<u128>,
 }
 
 impl Solution {
     pub fn trivial(num_orders: usize) -> Self {
         Solution {
-            prices: vec![],
+            prices: HashMap::new(),
             executed_buy_amounts: vec![0; num_orders],
             executed_sell_amounts: vec![0; num_orders],
         }
     }
 
+    pub fn max_token(&self) -> Option<u16> {
+        self.prices.keys().max().copied()
```

Comment:
Gooooooorgeous

Reply:
I still find it weird that max_tokens is optional. Shouldn't it return 0 if the map is empy?

Reply:
But that would suggest token id 0 is valid...

Reply:
I think `Some(0)` and `None` are slightly different here:
- `Some(0)` produces a price vector with 1 price.
- `None` produces an empty price vector.